### PR TITLE
feat(apigateway): add useDefaultMethodResponses property for StepFunctionsIntegrations

### LINKED ISSUE #27520: apigateway: StepFunctionsIntegrations add default methodResponses
### Describe the feature

When StepFunctionsIntegrations is used, it creates methodResponses for 200, 400 and 500 and it's not possible to override this behaviour.

### Use Case

When custom responses are provided they may clash with default ones.
For example one may want setup a different responseTemplate from the one provided.
For example I have a state machine that returns an URL and my responseTemplate set the http status to 302.

### Proposed Solution

Adding a `useDefaultMethodResponses?: boolean` prop (default true) to the `StepFunctionsExecutionIntegrationOptions` interface to avoid adding default responses:

https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-apigateway/lib/integrations/stepfunctions.ts#L160

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.100

### Environment details (OS name and version, etc.)

MacOs Sonoma

### PR DESCRIPTION
**Pull request by go-to-k**
In `StepFunctionsIntegration`, method response models (`methodResponses`) with 200, 400 and 500 status codes are added by default.

This PR adds `useDefaultMethodResponses` property not to add default method response models by set it to `false`.

Closes #27520.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Nice 👍
Just some documentation and test minor-changes needed in my opinion.

**Comment by go-to-k** [comment]
@lpizzinidev 

Thanks, I changed. Please review them again.

File: packages/aws-cdk-lib/aws-apigateway/test/stepfunctions-api.test.ts [hunk 1]
```diff
@@ -157,6 +157,75 @@ describe('Step Functions api', () => {
...
+              },
+              ':apigateway:',
+              {
+                Ref: 'AWS::Region',
+              },
+              ':states:action/StartSyncExecution',
+            ],
+          ],
+        },
+        PassthroughBehavior: 'NEVER',
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-apigateway/test/stepfunctions-api.test.ts:224]:
```suggestion
```
Let's match only on `HttpMethod` and `MethodResponses` to keep this concise.

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks, just a minor adjustment and it will be good to go for me.

**Comment by go-to-k** [comment]
Thanks, I changed it.

**Review by scanlonp — DISMISSED** [review]
Any reason to not have an integ test for the `StepFunctionsRestApi` invocation? Otherwise looks good to go.

**Comment by go-to-k** [comment]
@scanlonp 

I was satisfied with just the `StepFunctionsIntegration.startExecution` test. I added that integ test.

I did not want to interfere with the behavior of the existing StepFunctionsRestApi test with `useDefaultMethodResponses` is true as a default value, so I added a new test.

**Review by scanlonp — APPROVED** [review]
Great, thanks for the new integ test. The optional parameter is a good way to disable automatic behavior, and the tests cover both the template validation and deploy success.

**Pull request merged**